src/Termonad/Term.hs

```diff
@@ -425,64 +439,96 @@ setFocusOn tmStateAppWin vteTerm = do
+        termShellPid <- launchShell vteTerm maybeCurrDir
+        tmTerm <- newTMTerm scrolledWin vteTerm termShellPid
+        containerAdd scrolledWin vteTerm
+        pure (scrolledWin, vteTerm, tmTerm)
+
+  -- Create the left and right terminals
+  (scrolledWinR, vteTermR, tmTermR) <- createTerm
+  (scrolledWinL, vteTermL, tmTermL) <- createTerm
+
+  -- Create the paned window container add the two terminals to it
+  paned <- Gtk.panedNew OrientationHorizontal
+  Gtk.panedSetWideHandle paned True
+  Gtk.panedPack1 paned scrolledWinL True True
+  Gtk.panedPack2 paned scrolledWinR True True
+  Gtk.widgetShowAll paned
 
   -- Create the GTK widget for the Notebook tab
   (tabLabelBox, tabLabel, tabCloseButton) <- createNotebookTabLabel
 
   -- Create notebook state
-  let notebookTab = createTMNotebookTab tabLabel scrolledWin tmTerm
+  let notebookTab = createTMNotebookTab tabLabel paned tmTermL tmTermR
 
   -- Add the new notebooktab to the notebook.
   addPage mvarTMState notebookTab tabLabelBox
 
   -- Setup the initial label for the notebook tab.  This needs to happen
   -- after we add the new page to the notebook, so that the page can get labelled
-  -- appropriately.
-  relabelTab (tmNotebook currNote) tabLabel scrolledWin vteTerm
+  -- appropriately. We use the left terminal's title because it is the left
+  -- terminal which initially has the focus.
+  relabelTab (tmNotebook currNote) tabLabel paned vteTermL
 
   -- Connect callbacks
   void $ onButtonClicked tabCloseButton $ termClose notebookTab mvarTMState
-  void $ onTerminalWindowTitleChanged vteTerm $ do
+  void $ onTerminalWindowTitleChanged vteTermL $ do
+    -- TODO: use the title of the focused pane
     TMState{tmStateNotebook} <- readMVar mvarTMState
     let notebook = tmNotebook tmStateNotebook
-    relabelTab notebook tabLabel scrolledWin vteTerm
-  void $ onWidgetKeyPressEvent vteTerm $ handleKeyPress mvarTMState
-  void $ onWidgetKeyPressEvent scrolledWin $ handleKeyPress mvarTMState
-  void $ onWidgetButtonPressEvent vteTerm $ handleMousePress vteTerm
-  void $ onTerminalChildExited vteTerm $ \_ -> termExit notebookTab mvarTMState
-
-  -- Put the keyboard focus on the term
-  setFocusOn tmStateAppWin vteTerm
+    relabelTab notebook tabLabel paned vteTermL
+  for_ @[Terminal] [vteTermL, vteTermR] $ \vteTerm -> do
+    void $ onWidgetKeyPressEvent vteTerm $ handleKeyPress mvarTMState
+    void $ onWidgetButtonPressEvent vteTerm $ handleMousePress vteTerm
+    void $ onTerminalChildExited vteTerm $ \_ -> termExit notebookTab mvarTMState
+  for_ @[ScrolledWindow] [scrolledWinL, scrolledWinR] $ \scrolledWin -> do
+    void $ onWidgetKeyPressEvent scrolledWin $ handleKeyPress mvarTMState
+
+  void $ onWidgetFocusInEvent vteTermL $ \_ -> do
+    modifyMVar_ mvarTMState $ \tmState -> do
+      pure $ tmState & lensTMStateNotebook
+                     . traversalTMNotebookFocusedTab
+                     . lensTMNotebookTabFocusIsOnLeft .~ True
```

Comment:
I used to use `hindent`, so the style I use sort of comes from that, but I don't have a strong opinion on code style.  At this point multiple people have touched the code-base, so I don't think there is a single style used absolutely everywhere.

I think your code here is fine as-is.

At some point I guess it would be nice to pick a code formatter and set it up to run automatically.